packages/noya-state/src/reducers/pointReducer.ts

```diff
@@ -41,6 +56,54 @@ export function pointReducer(
         curvePoint.cornerRadius = Math.max(0, newValue);
       });
     }
+    case 'selectPoint': {
+      const [, rawPoint, mode] = action;
+      const pageIndex = getCurrentPageIndex(state);
+      const layerIndexPaths = getSelectedLayerIndexPaths(state);
+      const boundingRects = getBoundingRectMap(
+        getCurrentPage(state),
+        state.selectedObjects,
+        {
+          clickThroughGroups: true,
+          includeArtboardLayers: false,
+          includeHiddenLayers: false,
+        },
+      );
+
+      return produce(state, (draft) => {
+        layerIndexPaths.forEach((indexPath) => {
+          const page = draft.sketch.pages[pageIndex];
+          const layer = Layers.access(page, indexPath);
+          const pointList = draft.selectedPointLists[layer.do_objectID];
+          const boundingRect = boundingRects[layer.do_objectID];
+          if (!Layers.isPointsLayer(layer)) return;
+
+          layer.points.forEach((point, index) => {
+            const decodedPoint = decodeCurvePoint(point, boundingRect);
+            if (isPointInRange(decodedPoint.point, rawPoint) || !pointList) {
+              if (pointList.indexOf(index) > -1) {
+                mode === 'replace'
```

Comment:
Would these ternary expressions read better if they were switch statements?

Reply:
Yes! We shouldn't use assignment expressions when returning. Looks like there's an eslint rule I can add for this: https://eslint.org/docs/rules/no-return-assign

Reply:
The more eslint rules the better! Let's add it!

Reply:
Outer ternary expressions refactored to switch statements.